mlrun/api/main.py

```diff
@@ -155,23 +161,29 @@ async def startup_event():
     await _initialize_singletons()
 
     # periodic cleanup is not needed if we're not inside kubernetes cluster
-    if get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster():
+    if (
+        get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster()
+        and config.httpdb.state == mlrun.api.schemas.APIStates.online
+    ):
```

Comment:
If the service starts up in `offline` or in the middle of an upgrade, these will not get started. However, when migration is complete, nothing takes care of starting them - need to take care of that at the end of migrations.